### PR TITLE
docs - screenshot directive (and management command) can take screenshot as though the device was registered

### PR DESCRIPTION
**Pull request by MCGallaspy**
Because previously it couldn't. Does this by creating an associated Zone and DeviceZone for the Device, so that `Device.is_registered()` returns `True`.

Also touched up `sphinx-docs/screenshot.py` a little bit while I was in there, and added the appropriate option to a couple of screenshots.

### CODE REVIEW

File: sphinx-docs/screenshot.py [hunk 1]
```diff
@@ -283,21 +281,33 @@ def _command_handler(self, commands):
...
+            new_arg["focus"]["selector"] = self.focus_selector
+            new_arg["focus"]["styles"] = FOCUS_CSS_STYLES
+            new_arg["notes"] = self.focus_annotation if hasattr(self, "focus_annotation") else ""
+        new_arg["registered"] = self.options["registered"]
+        return new_arg
+
+    def _make_image_node(self):
+        """Make an image node by safely calling Image.run (i.e. ensure the file exists)."""
         self.arguments.append(os.path.join("/", SS_DUMP_DIR, self.filename+".png"))
         open(os.path.join(OUTPUT_PATH, self.filename+".png"), 'w').close()
```
Comment (aronasorman) [inline sphinx-docs/screenshot.py]: Hmm strange, we open the file and close it immediately? Is this a `touch`? Think we have an idiom for that already from the previous review.

Comment (aronasorman) [inline sphinx-docs/screenshot.py]: Also, spaces around the `+`.

File: sphinx-docs/screenshot.py [hunk 2]
```diff
@@ -203,6 +203,7 @@ class Screenshot(Image):
     option_spec['user-role'] = _parse_user_role
     option_spec['navigation-steps'] = _parse_nav_steps
     option_spec['focus'] = _parse_focus
+    option_spec['registered'] = lambda x: directives.choice(x, ("true", "false"))
```
Comment (aronasorman) [inline sphinx-docs/screenshot.py]: Interesting, so this is enforced by sphinx?

File: sphinx-docs/screenshot.py (hunk 1 above)
Comment (MCGallaspy) [inline sphinx-docs/screenshot.py]: It is indeed a `touch`. `utime` doesn't create files (just updates access time). This seems to be a standard way to do it.

File: sphinx-docs/screenshot.py (hunk 2 above)
Comment (MCGallaspy) [inline sphinx-docs/screenshot.py]: Do you mean are the values "true" and "false" enforced by Sphinx? Yes. It'll throw an error (without affecting other screenshots) if the option is specified incorrectly.

Comment (MCGallaspy) [inline sphinx-docs/screenshot.py]: However, failure to specify will default to `False`, which was the behavior before these changes -- screenshots were taken as though the device was not registered. In other words, these changes don't backwardly break anything.

**Pull request merged**